cmd/snap-update-ns/entry.go

```diff
@@ -104,3 +104,48 @@ func XSnapdGid(e *mount.Entry) (gid uint64, err error) {
 	}
 	return 0, nil
 }
+
+// XSnapdEntryID returns the identifier of a given mount enrty.
+//
+// Identifiers are kept in the x-snapd.id mount option. The value is a string
+// that identifies a mount entry and is stable across invocations of snapd (it
+// is a hash of some sort).
+func XSnapdEntryID(e *mount.Entry) string {
+	for _, opt := range e.Options {
+		if strings.HasPrefix(opt, "x-snapd.id=") {
+			kv := strings.SplitN(opt, "=", 2)
+			return kv[1]
+		}
+	}
+	return ""
+}
+
+// XSnapdParentID returns the identifier of the parent mount entry.
+//
+// Parent identifiers are kept in the x-snapd.parent-id mount option. The value
+// is a string that identifies a mount entry and is stable across invocations
+// of snapd (it is a hash of some sort).
+func XSnapdParentID(e *mount.Entry) string {
+	for _, opt := range e.Options {
+		if strings.HasPrefix(opt, "x-snapd.parent-id=") {
+			kv := strings.SplitN(opt, "=", 2)
+			return kv[1]
+		}
+	}
+	return ""
+}
+
+// XSnapdSynthetic returns true of a given mount entry is synthetic.
+//
+// Synthetic mount entries are created by snap-update-ns itself, separately
+// from what snapd instructed. Such entries are needed to make other things
+// possible.  They are identified by having the "x-snapd.synthetic" mount
+// option.
+func XSnapdSynthetic(e *mount.Entry) bool {
```

Comment:
Following the same idea as above, maybe this could be "mount.Entry.OptBool()" ? So something like http://paste.ubuntu.com/25917146/ ?

Reply:
Yep, good idea.

Reply:
Done